p2p/conn/connection.go

```diff
@@ -172,7 +189,11 @@ func (c *MConnection) OnStart() error {
 	}
 	c.quit = make(chan struct{})
 	c.flushTimer = cmn.NewThrottleTimer("flush", c.config.FlushThrottle)
-	c.pingTimer = cmn.NewRepeatTimer("ping", pingTimeout)
+	c.pingTimer = cmn.NewRepeatTimer("ping", c.config.PingInterval)
+	c.pongTimer = time.NewTimer(c.config.PongTimeout)
+	// we start timer once we've send ping; needed here because we use start
+	// listening in recvRoutine
+	_ = c.pongTimer.Stop()
```

Comment:
seems bad - what if the timer fires before we call stop?  maybe we can just create it the first time we need it ?  Otherwise we probably need the 

```
if !t.Stop() {
	<-t.C
}
```

here

Reply:
hmm. it could if pongTimeout is 0 or 1 ms... but it's 45 sec by default

Reply:
but what about in tests when its ms ? could be a source of non-deterministic failures :(

Reply:
sure, you're right. I'll fix. sorry